pkg/models/types.go

```diff
@@ -271,18 +279,260 @@ type AuditResponse struct {
 	Meta   *PaginationMeta `json:"paginationmeta"`
 }
 
+type ObligationType struct {
+	Id   int64  `gorm:"primary_key"`
+	Type string `gorm:"unique;not null"`
+}
+
+type ObligationClassification struct {
+	Id             int64  `gorm:"primary_key"`
+	Classification string `gorm:"unique;not null"`
+	Color          string `gorm:"unique; not null"`
+}
+
 // Obligation represents an obligation record in the database.
 type Obligation struct {
-	Id             int64  `gorm:"primary_key" json:"id" example:"147"`
-	Topic          string `gorm:"unique" json:"topic" example:"copyleft"`
-	Type           string `json:"type" enums:"obligation,restriction,risk,right" example:"risk"`
-	Text           string `json:"text" example:"Source code be made available when distributing the software."`
-	Classification string `json:"classification" enums:"green,white,yellow,red" example:"green"`
-	Modifications  bool   `json:"modifications" example:"true"`
-	Comment        string `json:"comment"`
-	Active         bool   `json:"active"`
-	TextUpdatable  bool   `json:"text_updatable" example:"true"`
-	Md5            string `gorm:"unique" json:"-"`
+	Id                         int64                     `gorm:"primary_key"`
+	Topic                      *string                   `gorm:"unique;not null"`
+	Text                       *string                   `gorm:"not null"`
+	Modifications              *bool                     `gorm:"not null;default:false"`
+	Comment                    *string                   `gorm:"not null;default:''"`
+	Active                     *bool                     `gorm:"not null;default:true"`
+	TextUpdatable              *bool                     `gorm:"not null;default:false"`
+	Md5                        string                    `gorm:"unique;not null"`
+	ObligationClassificationId int64                     `gorm:"not null"`
+	ObligationTypeId           int64                     `gorm:"not null"`
+	Licenses                   []*LicenseDB              `gorm:"many2many:obligation_licenses;"`
+	Type                       *ObligationType           `gorm:"foreignKey:ObligationTypeId"`
+	Classification             *ObligationClassification `gorm:"foreignKey:ObligationClassificationId"`
+}
+
+func (o *Obligation) BeforeCreate(tx *gorm.DB) (err error) {
```

Comment:
I would highly recommend moving all functions out of this file. Probably in the same package but a different file. It somewhat undermines the structure of the project.

Not a blocker.

Reply:
Yes. What do you think about moving a GORM struct and all it's methods into separate files? Like one for licenses, one for obligations etc.
I'll do it in another PR, this one's already big.

Reply:
This is a good idea, under `models` package, you can separate them into differnt files.